x-pack/packages/kbn-entities-schema/src/schema/common.ts

```diff
@@ -93,7 +93,30 @@ export const metadataSchema = z
     destination: metadata.destination ?? metadata.source,
     limit: metadata.limit ?? 1000,
   }))
-  .or(z.string().transform((value) => ({ source: value, destination: value, limit: 1000 })));
+  .or(z.string().transform((value) => ({ source: value, destination: value, limit: 1000 })))
+  .superRefine((value, ctx) => {
+    if (value.limit < 0) {
+      ctx.addIssue({
+        path: ['limit'],
+        code: z.ZodIssueCode.custom,
+        message: 'limit should be greater than 1',
+      });
+    }
+    if (value.source.length < 0) {
+      ctx.addIssue({
+        path: ['source'],
+        code: z.ZodIssueCode.custom,
+        message: 'source should not be empty',
+      });
+    }
+    if (value.destination.length < 0) {
```

Comment:
```suggestion
    if (value.limit < 1) {
      ctx.addIssue({
        path: ['limit'],
        code: z.ZodIssueCode.custom,
        message: 'limit should be greater than 1',
      });
    }
    if (value.source.length === 0) {
      ctx.addIssue({
        path: ['source'],
        code: z.ZodIssueCode.custom,
        message: 'source should not be empty',
      });
    }
    if (value.destination.length === 0) {
```

Reply:
Wow... I really effed up the logic on that. Not sure what I was thinking.

Reply:
I added a test to ensure this is correct.